pluginutils.cpp

```diff
@@ -236,4 +237,33 @@ namespace pluginutils
 		return false;
 	}
 
+	int SetAmxString(AMX* amx, cell params, std::string_view str) 
+	{
+		cell* dest = reinterpret_cast<cell*>(amx->base + static_cast<int>(reinterpret_cast<AMX_HEADER*>(amx->base)->dat + params));
```

Comment:
This would fail horribly if the user passes an invalid AMX address from the script side (e.g. because of bad use of `#emit`).

Reply:
Oh, okey. I can use amx_GetAddr / amx_GetString. Is this a better solution?